modules/plugins/youtube.go

```diff
@@ -199,93 +218,230 @@ func (yt *YouTube) actionChannel(args []string, in *discordgo.Message, out **dis
+	}
 
-	*out = yt.newMsg("bot.arguments.invalid")
+	// search channel
+	yc, err := yt.searchQuerySingle(args[2:len(args)-1], "channel")
+	if err != nil {
+		yt.logger().Error(err)
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
+
+	if yc == nil {
+		*out = yt.newMsg("plugins.youtube.channel-not-found")
+		return yt.actionFinish
+	}
+
+	content := DB_Youtube_Content_Channel{
+		ID:   yc.Id.ChannelId,
+		Name: yc.Snippet.ChannelTitle,
+	}
+
+	// Very few channels only have snippet.ChannelID
+	// Maybe it's youtube API bug.
+	if content.ID == "" {
+		content.ID = yc.Snippet.ChannelId
+	}
+
+	if content.ID == "" || content.Name == "" {
+		*out = yt.newMsg("plugins.youtube.channel-not-found")
+		return yt.actionFinish
+	}
+
+	// fill db entry
+	entry := DB_Youtube_Entry{
+		ServerID:                dc.GuildID,
+		ChannelID:               dc.ID,
+		NextCheckTime:           time.Now().Unix(),
+		LastSuccessfulCheckTime: time.Now().Unix(),
+
+		ContentType: "channel",
+		Content:     content,
+	}
+
+	// insert entry into the db
+	_, err = yt.createEntry(entry)
+	if err != nil {
+		yt.logger().Error(err)
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
+
+	yt.quota.IncEntryCount()
+
+	*out = yt.newMsg("Added youtube channel `" + yc.Snippet.ChannelTitle + "` to the discord channel <#" + dc.ID + ">")
 	return yt.actionFinish
 }
 
+// _yt channel delete <channel id> <discord channel>
 func (yt *YouTube) actionDeleteChannel(args []string, in *discordgo.Message, out **discordgo.MessageSend) youtubeAction {
-	if len(args) < 4 {
+	if len(args) < 3 {
 		*out = yt.newMsg("bot.arguments.too-few")
 		return yt.actionFinish
 	}
 
-	// _yt video delete <video id/link> <discord channel>
+	if helpers.IsMod(in) == false {
+		*out = yt.newMsg("mod.no_permission")
+		return yt.actionFinish
+	}
 
-	*out = yt.newMsg("bot.arguments.invalid")
+	n, err := yt.deleteEntry(args[2])
+	if err != nil {
+		yt.logger().Error(err)
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
+
+	if n < 1 {
+		*out = yt.newMsg("plugins.youtube.channel-delete-not-found-error")
+		return yt.actionFinish
+	}
+
+	yt.quota.DecEntryCount()
+
+	*out = yt.newMsg("Delete channel, ID: " + args[2])
 	return yt.actionFinish
 }
 
+// _yt channel list
 func (yt *YouTube) actionListChannel(args []string, in *discordgo.Message, out **discordgo.MessageSend) youtubeAction {
+	if helpers.IsMod(in) == false {
+		*out = yt.newMsg("mod.no_permission")
+		return yt.actionFinish
+	}
 
-	// _yt video list
+	ch, err := helpers.GetChannel(in.ChannelID)
+	if err != nil {
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
 
-	*out = yt.newMsg("bot.arguments.invalid")
+	entries, err := yt.readEntries(map[string]interface{}{
+		"content_type": "channel",
+		"server_id":    ch.GuildID,
+	})
+	if err != nil {
+		yt.logger().Error(err)
+		*out = yt.newMsg(err.Error())
+		return yt.actionFinish
+	}
+
+	if len(entries) < 1 {
+		*out = yt.newMsg("No entries")
```

Comment:
Please add this to the i18n json for easier editing

Reply:
changed